tensorbay/opendataset/BDD100K/loader.py

```diff
@@ -279,6 +361,73 @@ def _merge_label(source_label_contents: List[List[Dict[str, Any]]]) -> Dict[str,
     return label_contents
 
 
+def _get_instance_mask(filename: str, original_mask_path: str, mask_directory: str) -> InstanceMask:
+    mask_path = os.path.join(mask_directory, f"{filename}.png")
+    mask_info = _save_and_get_mask_info(
+        os.path.join(original_mask_path, f"{filename}.png"),
+        mask_path,
+        os.path.join(mask_directory, f"{filename}.json"),
+        "ins",
+    )
+
+    ins_mask = InstanceMask(mask_path)
+    ins_mask.all_attributes = mask_info["all_attributes"]
+    return ins_mask
+
+
+def _get_panoptic_mask(filename: str, original_mask_path: str, mask_directory: str) -> PanopticMask:
+    mask_path = os.path.join(mask_directory, f"{filename}.png")
+    mask_info = _save_and_get_mask_info(
+        os.path.join(original_mask_path, f"{filename}.png"),
+        mask_path,
+        os.path.join(mask_directory, f"{filename}.json"),
+        "pan",
+    )
+
+    pan_mask = PanopticMask(mask_path)
+    pan_mask.all_category_ids = mask_info["all_category_ids"]
+    pan_mask.all_attributes = mask_info["all_attributes"]
+    return pan_mask
+
+
+def _save_and_get_mask_info(
+    original_mask_path: str, mask_path: str, mask_info_path: str, seg_type: str
+) -> Dict[str, Any]:
+    if not os.path.exists(mask_path):
+        mask = np.array(Image.open(original_mask_path))
+        all_attributes = {}
+        if seg_type == "pan":
+            all_category_ids = {}
+        for instance_info in set(map(tuple, np.reshape(mask, (-1, 4)))):
```

Comment:
what is the `set` for?

Reply:
Is this map to tuple necessary?

Reply:
Use `set` to get all different pixels
`np.ndarray` is unhashable, so I must map the pixels to `tuple`